server/repository/src/db_diesel/temperature_breach_row.rs

```diff
@@ -0,0 +1,141 @@
+use super::{
+    location_row::location, sensor_row::sensor, store_row::store,
+    temperature_breach_row::temperature_breach::dsl as temperature_breach_dsl, StorageConnection,
+};
+
+use crate::repository_error::RepositoryError;
+
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel_derive_enum::DbEnum;
+use serde::{Deserialize, Serialize};
+
+table! {
+    temperature_breach (id) {
+        id -> Text,
+        duration -> Integer,
+        #[sql_name = "type"] type_ -> crate::db_diesel::temperature_breach_row::TemperatureBreachRowTypeMapping,
```

Comment:
Technically, `ColdCumulative` will never be saved in TemperatureBreach table, I think it's a good idea to make a comment somewhere (either here or a smart comment in schema). Btw, i am not sure if we show cumulative in mobile but in cold chain app it's based on the display windows of temperature charts. 

>  I've tried to share the enum breach type with the config, but there may be a better way to doing it?

I think it's great thank you (would assume it would also be a shared type in postgres schema)

Reply:
Not sure that I understand why `ColdCumulative` would never be saved? For some types of FridgeTags, the sensor itself _only_ generates cumulative breaches, and the only way to generate consecutive breaches is to calculate them... mSupply desktop shows both.